cpp/src/arrow/array/validate.cc

```diff
@@ -464,8 +465,8 @@ struct ValidateArrayImpl {
     return data.buffers[index] != nullptr && data.buffers[index]->address() != 0;
   }
 
-  Status RecurseInto(const ArrayData& related_data) {
-    ValidateArrayImpl impl{related_data, full_validation};
+  Status RecurseInto(const ArrayData& related_data, bool nullable = true) {
```

Comment:
I would rather make the argument mandatory, so that we don't forget to pass it.
```suggestion
  Status RecurseInto(const ArrayData& related_data, bool nullable) {
```